prov/util/src/util_av.c

```diff
@@ -415,23 +426,32 @@ static int util_av_init(struct util_av *av, const struct fi_av_attr *attr,
 		       "FI_SOURCE requested, hash size %zu\n", av->hash.total_count);
 	}
 
-	av->data = malloc((av->count * util_attr->addrlen) +
+	av->addr = malloc((av->count * util_attr->addrlen) +
 			  (av->hash.total_count * sizeof(*av->hash.table)));
-	if (!av->data)
+	if (!av->addr)
 		return -FI_ENOMEM;
 
+	av->data = calloc(av->count, sizeof *(av->data));
+	if (!av->data) {
+		ret = -FI_ENOMEM;
+		goto err;
+	}
```

Comment:
We should probably make the data allocation optional - maybe controlled through an internal flag?